manifests/postgres_exporter.pp

```diff
@@ -0,0 +1,209 @@
+# Class: prometheus::postgres_exporter
+#
+# This module manages prometheus node postgres_exporter
+#
+# Parameters:
+#  [*arch*]
+#  Architecture (amd64 or i386)
+#
+#  [*bin_dir*]
+#  Directory where binaries are located
+#
+#  [*addr*]
+#  Array of address of one or more redis nodes. Defaults to redis://localhost:6379
+#
+#  [*download_extension*]
+#  Extension for the release binary archive
+#
+#  [*download_url*]
+#  Complete URL corresponding to the where the release binary archive can be downloaded
+#
+#  [*download_url_base*]
+#  Base URL for the binary archive
+#
+#  [*extra_groups*]
+#  Extra groups to add the binary user to
+#
+#  [*extra_options*]
+#  Extra options added to the startup command
+#  For a full list of the exporter's supported extra options
+#  please refer to https://github.com/wrouesnel/postgres_exporter
+#
+#  [*group*]
+#  Group under which the binary is running
+#
+#  [*init_style*]
+#  Service startup scripts style (e.g. rc, upstart or systemd)
+#
+#  [*install_method*]
+#  Installation method: url or package (only url is supported currently)
+#
+#  [*manage_group*]
+#  Whether to create a group for or rely on external code for that
+#
+#  [*manage_service*]
+#  Should puppet manage the service? (default true)
+#
+#  [*manage_user*]
+#  Whether to create user or rely on external code for that
+#
+#  [*os*]
+#  Operating system (linux is the only one supported)
+#
+#  [*package_ensure*]
+#  If package, then use this for package ensure default 'latest'
+#
+#  [*package_name*]
+#  The binary package name - not available yet
+#
+#  [*purge_config_dir*]
+#  Purge config files no longer generated by Puppet
+#
+#  [*restart_on_change*]
+#  Should puppet restart the service on configuration change? (default true)
+#
+#  [*service_enable*]
+#  Whether to enable the service from puppet (default true)
+#
+#  [*service_ensure*]
+#  State ensured for the service (default 'running')
+#
+#  [*service_name*]
+#  Name of the node exporter service (default 'postgres_exporter')
+#
+#  [*user*]
+#  User which runs the service
+#
+#  [*version*]
+#  The binary release version
+#
+#  [*postgres_user*]
+#  User to authenticate against postgres
+#
+#  [*postgres_pass*]
+#  Password to authenticate against postgres
+#
+#  [*postgres_auth_method*]
+#  method for presenting username and password to the exporter
+#  This can be file, or env (default 'env')
+#
+#  [*data_source_uri*]
+#  Uri on howto connect to the database
+
+class prometheus::postgres_exporter (
+  String[1] $download_extension,
+  String[1] $download_url_base,
+  Array[String] $extra_groups,
```

Comment:
Can you please update this to `Array[String[1]]`? That will enforce a minimum string length of 1.

Reply:
ok. done